src/NHapi.Base/SupportClass.cs

```diff
@@ -1,3268 +1,3486 @@
+        /// run method to be called in that separately executing thread.
+        /// </summary>
+        void Run();
+    }
+
+    /*******************************/
+
+    /// <summary>
+    /// This interface will manage errors during the parsing of a XML document.
+    /// </summary>
+    public interface IXmlSaxErrorHandler
+    {
+        /// <summary>
+        /// This method manage an error exception occurred during the parsing process.
+        /// </summary>
+        /// <param name="exception">The exception thrown by the parser.</param>
+        void error(XmlException exception);
+
+        /// <summary>
+        /// This method manage a fatal error exception occurred during the parsing process.
+        /// </summary>
+        /// <param name="exception">The exception thrown by the parser.</param>
+        void fatalError(XmlException exception);
+
+        /// <summary>
+        /// This method manage a warning exception occurred during the parsing process.
+        /// </summary>
+        /// <param name="exception">The exception thrown by the parser.</param>
+        void warning(XmlException exception);
+    }
+
+    /*******************************/
+
+    /// <summary>
+    /// This class is used to encapsulate a source of Xml code in an single class.
+    /// </summary>
+    public class XmlSourceSupport
+    {
+        private Stream bytes;
+        private StreamReader characters;
+        private string uri;
+
+        /// <summary>
+        /// Constructs an empty XmlSourceSupport instance.
+        /// </summary>
+        public XmlSourceSupport()
+        {
+            bytes = null;
+            characters = null;
+            uri = null;
+        }
+
+        /// <summary>
+        /// Constructs a XmlSource instance with the specified source System.IO.Stream.
+        /// </summary>
+        /// <param name="stream">The stream containing the document.</param>
+        public XmlSourceSupport(Stream stream)
+        {
+            bytes = stream;
+            characters = null;
+            uri = null;
+        }
+
+        /// <summary>
+        /// Constructs a XmlSource instance with the specified source System.IO.StreamReader.
+        /// </summary>
+        /// <param name="reader">The reader containing the document.</param>
+        public XmlSourceSupport(StreamReader reader)
+        {
+            bytes = null;
+            characters = reader;
+            uri = null;
+        }
+
+        /// <summary>
+        /// Construct a XmlSource instance with the specified source Uri string.
+        /// </summary>
+        /// <param name="source">The source containing the document.</param>
+        public XmlSourceSupport(string source)
+        {
+            bytes = null;
+            characters = null;
+            uri = source;
+        }
+
+        /// <summary>
+        /// Represents the source Stream of the XmlSource.
+        /// </summary>
+        public Stream Bytes

```

Comment:
get rid of fields and just use propers for these obvious ones.

Reply:
will log a separate issue to refactor this `SupportClass`